### DISCUSSION TITLE
jupyter lab的方便功能🖊️ ✏️ 📝⏪

### DISCUSSION THREAD

**Discussion by enoch-lightning** [1101-GS3073-C/Python-Community-GS3073-C#264] [General]
:two: :zero: :two: :two:/:three:/:one: :pushpin: :star:
💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 💪 
# 📢 好康報你知!
![image](https://user-images.githubusercontent.com/86505454/156556481-a3aca143-2f23-4c9a-9e7e-34971de2bb5e.png)

使用jupyter lab有些很方便的功能，目前整理出3個。
- `dir()`:將此模組或函式庫能使用的函式(function)、變數、類別的實例(instance)列舉出來
- `.+tab`:在模組或函式庫名稱後加上`.`，再按tab，jupyter lab會幫我們整理出能用的函式(function)、變數、類別的實例(instance)
- `函式(function_name())+(shift+tab)`:若想在jupyter lab看一個函式的docstring(document string)，則可以在打完函式e.g.`input()`後，將游標移到`input()`上，按下shift+tab(先按住shift不要放，再按tab)，便可看到`input`的docstring。`input()`的docstring如下。
![image](https://user-images.githubusercontent.com/86505454/156149161-2dd12c1f-b3e4-47bd-abfa-36aee022cb2c.png)
問題：我看不懂`If the user hits EOF (*nix: Ctrl-D, Windows: Ctrl-Z+Return), raise EOFError.
On *nix systems, readline is used if available.`在寫什麼。

**Comment by timchen1216** [comment]
補充jupyter快捷鍵
https://www.kdzone.net/2017/08/jupyter-notebook.html

但是我要說 唯一支持vs code!!!
vs code好用多了
補上vs code快捷鍵
https://blog.poychang.net/vscode-shortcuts/

**Reply by Bill671** [reply]
大推vscode !! 還支援其他語言，讓跨語言撰寫不用安裝一堆idle

**Comment by enoch-lightning** [comment]
我來試試markdown(md)語法的一個功能，還是其實是html語法？
<s>abcdefghijklmnopqrstuv我是人類。</s>
我使用的語法是`<s>abcdefghijklmnopqrstuv我是人類。</s>`

**Comment by lx3211** [comment]
dir()好像不是單屬於Jupiter lab的語法喔

**Comment by NcuJerryKao** [comment]
為甚麼不直接用jupyter而要用網頁版的 難道有什麼不為人知的優點嗎
我還是比較喜歡用自己安裝的jupyter
